### PR TITLE
NGSTACK-853 impersonation module

### PR DESCRIPTION
**Pull request by PetraMiseta**
Added impersonation module in adminui and backoffice (+ necessary assets and event listener).

Installed version dev-NGSTACK-853 (updated site api version to 6)

### CODE REVIEW

File: assets/sass/_user_impersonation.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        font-weight: 600;
+        gap: 0.25rem;
+
+        &.btn:not(.btn.btn-link) {
+            @include media-breakpoint-down(sm) {
+                width: fit-content;
+            }
+        }
+    }
+}
```
Comment (pspanja) [inline assets/sass/_user_impersonation.scss:43]: Missing empty line at the end of file.

File: src/EventListener/Security/InteractiveLoginListener.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use function array_merge;
+use function array_unique;
+use function get_class;
+use function method_exists;
+use function sprintf;
+
+final class InteractiveLoginListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private UserRoleResolver $userRoleResolver,
```
Comment (pspanja) [inline src/EventListener/Security/InteractiveLoginListener.php:29]: This is not used.

File: assets/sass/_user_impersonation.scss (hunk 1 above)
Comment (PetraMiseta) [reply assets/sass/_user_impersonation.scss]: @pspanja added here https://github.com/netgen/media-site/pull/194/commits/67b25506a8e18fdbaa9423904b4fb46a20f5d94a

File: src/Admin/Menu/MainMenuBuilder.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
```
Comment (emodric) [inline src/Admin/Menu/MainMenuBuilder.php:1]: Added in error?

File: src/Security/UserRoleResolver.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+use Netgen\IbexaSiteApi\API\Values\Location;
+use Netgen\IbexaSiteApi\Core\Traits\SearchResultExtractorTrait;
+
+use function array_filter;
+use function array_map;
+use function array_unique;
+use function count;
+use function in_array;
+
+final class UserRoleResolver
```
Comment (emodric) [inline src/Security/UserRoleResolver.php:25]: Is this really related to the impersonation module?

File: config/app/app.yaml [hunk 5]
```diff
@@ -15,3 +15,19 @@ parameters:
 
     app.testing.site_domain: '%env(TEST_DOMAIN)%'
     ngsite.default.email_change.token_validity_time_minutes: 15
+
+    netgen_user_impersonation.default.content_type_identifier.user_groups: ['user_group']
+    netgen_user_impersonation.default.content_type_identifier.users: ['user']
+    netgen_user_impersonation.default.users.root_location_ids: [5]
+    netgen_user_impersonation.default.users.name_search_fields: ['first_name', 'last_name']
+    netgen_user_impersonation.default.excluded_user_groups:
+        - 13 # administrator users
```
Comment (emodric) [inline config/app/app.yaml:24]: Since these are ID's, they should go to `config/app/server`, since I see this as beeing modified based on project requirements.

File: config/app/app.yaml [hunk 6]
```diff
@@ -15,3 +15,19 @@ parameters:
...
+        - 44 # anonymous users
+
+    ngsite.default.roles_by_user_groups:
```
Comment (emodric) [inline config/app/app.yaml:27]: This config is also related to the comment left for `UserRoleResolver` comment.

File: src/Security/UserRoleResolver.php (hunk 4 above)
Comment (emodric) [reply src/Security/UserRoleResolver.php:25]: Besides, it is missing the login listener which actually runs the resolver.

File: src/Admin/Menu/MainMenuBuilder.php (hunk 3 above)
Comment (PetraMiseta) [reply src/Admin/Menu/MainMenuBuilder.php]: Yes, deleted here https://github.com/netgen/media-site/pull/194/commits/0c730f88e1cced16467dbb166e16624ac37fccc8

File: src/Security/UserRoleResolver.php (hunk 4 above)
Comment (PetraMiseta) [reply src/Security/UserRoleResolver.php]: I added it while we had the login listener in the project and forgot to remove it after I deleted the listener. Deleted now https://github.com/netgen/media-site/pull/194/commits/0c730f88e1cced16467dbb166e16624ac37fccc8

File: config/app/app.yaml (hunk 5 above)
Comment (PetraMiseta) [reply config/app/app.yaml]: Done - https://github.com/netgen/media-site/pull/194/commits/d4edcafa8f32db685ec444568b26293cf12ed005

**Review by emodric — APPROVED** [review]

**Review by pspanja — APPROVED** [review]

**Comment by emodric** [comment]
Thanks @PetraMiseta !
